store/tikv/prewrite.go

```diff
@@ -276,12 +277,16 @@ func (action actionPrewrite) handleSingleBatch(c *twoPhaseCommitter, bo *Backoff
 			// Check already exists error
 			if alreadyExist := keyErr.GetAlreadyExist(); alreadyExist != nil {
 				e := &tikverr.ErrKeyExist{AlreadyExist: alreadyExist}
-				return c.extractKeyExistsErr(e)
+				err = c.extractKeyExistsErr(e)
+				if err != nil {
+					atomic.StoreUint32(&c.prewriteFailed, 1)
+				}
```

Comment:
```suggestion
		        atomic.StoreUint32(&c.prewriteFailed, 1)
				return c.extractKeyExistsErr(e)
```
Seems the err must be non-nil, and you forget to return the error.

Reply:
oops...